cpp/daal/src/algorithms/tsne/tsne_gradient_descent_fpt.cpp

```diff
@@ -128,24 +129,24 @@ services::Status boundingBoxKernelImpl(DataType * posx, DataType * posy, const I
 
     //scale the maximum to get all points strictly in the bounding box
     if (box[1] >= 0.)
-        box[1] = (box[1] * (1. + 1e-3));
+        box[1] = (box[1] * (1. + boxEps));
     else
-        box[1] = (box[1] * (1. - 1e-3));
+        box[1] = (box[1] * (1. - boxEps));
     if (box[3] >= 0.)
-        box[3] = (box[3] * (1. + 1e-3));
+        box[3] = (box[3] * (1. + boxEps));
     else
-        box[3] = (box[3] * (1. - 1e-3));
+        box[3] = (box[3] * (1. - boxEps));
 
     //save results
     radius       = services::internal::max<cpu, DataType>(box[1] - box[0], box[3] - box[2]) * 0.5;
-    posx[nNodes] = (box[0] + box[1]) * 0.5;
-    posy[nNodes] = (box[2] + box[3]) * 0.5;
+    posX[nNodes] = (box[0] + box[1]) * 0.5;
+    posY[nNodes] = (box[2] + box[3]) * 0.5;
```

Comment:
`0.5` refers to `double` precision data type. Please consider the following:
```suggestion
    posY[nNodes] = (box[2] + box[3]) * static_cast<DataType>(0.5);
```